tests/vmi_multus_test.go

```diff
@@ -876,29 +887,197 @@ var _ = Describe("[Serial]SRIOV", func() {
 	})
 })
 
+var _ = Describe("[Serial]Macvtap", func() {
+	var err error
+	var virtClient kubecli.KubevirtClient
+	var macvtapLowerDevice string
+	var macvtapNetworkName string
+
+	tests.BeforeAll(func() {
+		virtClient, err = kubecli.GetKubevirtClient()
+		tests.PanicOnError(err)
+
+		macvtapLowerDevice = "eth0"
+		macvtapNetworkName = "net1"
+
+		// cleanup the environment
+		tests.BeforeTestCleanup()
+	})
+
+	BeforeEach(func() {
+		tests.EnableFeatureGate(virtconfig.MacvtapGate)
+	})
+
+	BeforeEach(func() {
+		result := virtClient.RestClient().
+			Post().
+			RequestURI(fmt.Sprintf(postUrl, tests.NamespaceTestDefault, macvtapNetworkName)).
+			Body([]byte(fmt.Sprintf(macvtapNetworkConf, macvtapNetworkName, tests.NamespaceTestDefault, macvtapLowerDevice, macvtapNetworkName))).
+			Do()
+		Expect(result.Error()).NotTo(HaveOccurred(), "A macvtap network named %s should be provisioned", macvtapNetworkName)
+	})
+
+	AfterEach(func() {
```

Comment:
I would suggest that you remove the `BeforeAll` and pretty much all the `AfterEach`. Then create one `BeforeEach` which calls the usual `BeforeTestCleanup` function like most other tests do. In case that we don't delete all network attachment definitions in the namespace in `BeforeTestCleanup` add  it there.

Reply:
Done.